adagrams/game.py

```diff
@@ -1,13 +1,129 @@
+        for i in range(freq):
+            letter_deck.append(letter)
+    
+    for i in range(10):
+        index = randint(0,len(letter_deck)-1)
+        letter_bank.append(letter_deck[index])
+        letter_deck.pop(index)
+
+    return letter_bank
+ 
 
 def uses_available_letters(word, letter_bank):
-    pass
+    filtered = []
+    letter_bank_count = {}
+    # Count letters in the bank, store in a dict
+    for letter in letter_bank:
+        letter_bank_count[letter] = letter_bank_count.get(letter,0) + 1
+
+
+
+    for letter in list(word.upper()):
+        if letter_bank_count.get(letter,0) > 0:
+            filtered.append(letter)  # Build filtered list and check availability
+            letter_bank_count[letter] -= 1 # decrease count in dict only
+        else:
+            return False
+
+   # Check if all letters were used
+    if filtered == list(word.upper()):
+        return True
+    else:
+        return False
+
+# rethinking on 9/19/2025 of the function uses_available_letters:
+# changes and reason :
+# I was building a separate filtered list. But I removed it bc the loop already checks if each letter is available in the letter bank, so if the loop finishes, we know the word can be formed.
+# updated function :
+# def uses_available_letters(word, letter_bank): 
+#     letter_bank_count = {} 
+#     for letter in letter_bank: 
+#         letter_bank_count[letter] = letter_bank_count.get(letter, 0) + 1 
+        
+#     for letter in word.upper(): 
+#         if letter_bank_count.get(letter, 0) == 0: 
+#             return False 
+#         letter_bank_count[letter] -= 1 
+        
+#     return True
+
+    
 
 def score_word(word):
-    pass
+    LETTER_SCORES = { "A": 1, "E": 1, "I": 1, "O": 1, "U": 1, "L": 1, "N": 1, "R": 1, "S": 1, "T": 1, 
+                     "D": 2, "G": 2, "B": 3, 
+                     "C": 3, "M": 3, "P": 3, 
+                     "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4, 
+                     "K": 5, 
+                     "J": 8, "X": 8, 
+                     "Q": 10, "Z": 10 
+    }
+    score = 0
+    if not word:
+        return score
+    else:
+        
+        for letter in word.upper():
+            score += LETTER_SCORES[letter]
+
+    if len(word) >= 7:
+        score += 8
+    return  score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    best_word = word_list[0]
+    score = score_word(word_list[0])
+
+    for i in range(1,len(word_list)): 
+        current_word = word_list[i]
+        current_score = score_word(current_word)
+
+
+        if current_score > score:
+            best_word = current_word
+            score = current_score
+
+        elif current_score == score:
+            if len(current_word) == 10 and len(best_word) != 10:
+                best_word = current_word
+            else:
+                if len(best_word) != 10 and len(current_word) < len(best_word):
+                    best_word = current_word
```

Comment:
This looks good! If you wanted to, you could throw the tie breaking logic into a helper function, but overall this is ok!